### PR TITLE
feat: support portal restTemplate Client connection pool config

### LINKED ISSUE #5199: Portal 模块 RestTemplate 支持配置 HTTP最大连接池和每个路由的连接大小
**你的特性请求和某个问题有关吗？请描述**

- 当前 portal 服务发现 调用Admin服务的时候，使用的是RestTemplate的连接池使用是默认的参数，默认是每个路由2，最大连接数20
 [RestTemplateFactory](https://github.com/apolloconfig/apollo/blob/master/apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/component/RestTemplateFactory.java#L61)
- 在公司内部对 Apollo OpenAPI的接口 修改配置接口 和 发布配置接口 进行压测时，发现最大QPS只有200，但是CPU和线程还都比较空闲，使用Arthas查看所有的线程都卡在获取HTTP连接上，通过增加RestTemplate 的 connectionManager的MaxTotal（最大连接数） 和  MaxPerRoute（每个路由的连接大小）的大小解决并发问题

**清晰简洁地描述一下你希望的解决方案**
希望 Portal 模块 支持配置 RestTemplate 的 connectionManager的MaxTotal（最大连接数） 和  MaxPerRoute（每个路由的连接大小）
```
PoolingHttpClientConnectionManager connectionManager = new PoolingHttpClientConnectionManager();
connectionManager.setMaxTotal(portalConfig.connectPoolMaxTotal()); //最大连接数
connectionManager.setDefaultMaxPerRoute(portalConfig.connectPoolMaxPerRoute()); //每个路由（域名）最大连接数
```
在 PortalConfig 新增配置属性
```
public int connectPoolMaxTotal() {
    return getIntProperty("api.pool.max.total", 200);
  }

  public int connectPoolMaxPerRoute() {
    return getIntProperty("api.pool.max.per.route", 20);
  }
```

**清晰简洁地描述一下这个特性的备选方案**

> 无

**其它背景**

在这里添加和这个特性请求有关的背景说明、截图

### PR DESCRIPTION
**Pull request by youngzil**
## What's the purpose of this PR

Portal 模块 RestTemplate 支持配置 HTTP最大连接池和每个路由的连接大小 详见 #5199

## Which issue(s) this PR fixes:
Fixes #5199

## Brief changelog
support portal restTemplate Client connection pool config

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Read the [Contributing Guide](https://github.com/apolloconfig/apollo/blob/master/CONTRIBUTING.md) before making this pull request.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit tests to verify the code.
- [x] Run `mvn clean test` to make sure this pull request doesn't break anything.
- [x] Update the [`CHANGES` log](https://github.com/apolloconfig/apollo/blob/master/CHANGES.md).

## Summary by CodeRabbit

- **New Features**
	- Added support for configuring a connection pool for the HTTP client, enhancing performance and efficiency in handling multiple connections.
	- Introduced new configuration methods to manage connection pooling parameters.

- **Documentation**
	- Updated changelog to reflect the addition of connection pool configuration support for the HTTP client.

### CODE REVIEW

File: apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/component/RestTemplateFactory.java [hunk 1]
```diff
@@ -59,8 +60,14 @@ public boolean isSingleton() {
   }
 
   public void afterPropertiesSet() throws UnsupportedEncodingException {
+
+    PoolingHttpClientConnectionManager connectionManager = new PoolingHttpClientConnectionManager();
+    connectionManager.setMaxTotal(portalConfig.connectPoolMaxTotal()); //最大连接数
+    connectionManager.setDefaultMaxPerRoute(portalConfig.connectPoolMaxPerRoute()); //每个路由（域名）最大连接数
```
Comment (nobodyiam) [inline apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/component/RestTemplateFactory.java:66]: Please provide your comments in English.

File: apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/component/config/PortalConfig.java [hunk 2]
```diff
@@ -173,6 +173,14 @@ public int connectionTimeToLive() {
     return getIntProperty("api.connectionTimeToLive", -1);
   }
 
+  public int connectPoolMaxTotal() {
+    return getIntProperty("api.pool.max.total", 20);
```
Comment (nobodyiam) [inline apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/component/config/PortalConfig.java:177]: What is the default configuration when no connection manager is specified?

Comment (youngzil) [reply apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/component/config/PortalConfig.java:177]: The default configuration of the connection pool is that the maximum number of connections is 20 and the number of connections per route is 2. This is consistent with the default value of this configuration.

File: apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/component/RestTemplateFactory.java (hunk 1 above)
Comment (youngzil) [reply apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/component/RestTemplateFactory.java:66]: Maybe I can remove this comment, I think the variable name is clear enough

**Comment by nobodyiam** [comment]
Please complete the CLA by following the instructions provided by the CLA bot.
Once done, you may proceed to refresh the codebase and update the CHANGES.md file.

**Comment by youngzil** [comment]
I have read the CLA Document and I hereby sign the CLA

**Pull request merged**